### ISSUE TITLE
Allow tests in `/src` directory

### ISSUE THREAD

**Issue by Nick-Mazuk** [gleam-lang/gleam#2660]
## Problem

It seems like tests are only allowed in the `/test` directory. This means that as a codebase scales to hundreds-thousands of files, the tests progressively get further from the related code.

Having worked in large codebases with tests both right next to the relevant code and in a separate top-level folder, I can say I'm far more productive when the test is right next to the code.

## Solutions

There's three solutions:

1. The test is in the same file as the code (e.g., Rust).
2. The test is in a file next to the code (e.g., Golang, TypeScript).

Either works.

### My preferred solution

My preference is to have tests in a separate file. So…

```txt
/src
  /hello.gleam
  /hello_test.gleam
```

This yields a few advantages over having tests in the same file:

- Test code is not intermixed with production code.
- As a code reviewer, I can immediately look at the tests.

**Comment by lpil** [comment]
Hello! You can do this today if you want, Gleam doesn't dictate anything about how tests are run and just calls your `PACKAGE_test.main` function for you.

**Comment by jayvanhu** [comment]
> Hello! You can do this today if you want, Gleam doesn't dictate anything about how tests are run and just calls your PACKAGE_test.main function for you when you run gleam test.

Hello :wave:, I'm not sure if I'm missing something, but just running `gleam test` for me with a test inside the `src/` directory doesn't work. I ran `gleam new foo`, then immediately moved the `foo_test.gleam` file to inside `src/` and ran `gleam test`, and it resulted in `0 tests`. Moving `foo_test.gleam` back into the `test/` directory and running `gleam test` again results in `1 test, 0 failures`. Is there something else I have to do other than `gleam test`? I tried `gleam test ./src` and that didn't work either. Would appreciate some help, I can't figure it out
